packages/web-components/fast-foundation/src/carousel/carousel.ts

```diff
@@ -0,0 +1,411 @@
+    public itemsChanged(): void {
+        if (this.items.length && this.basicPattern) {
+            this.generateSlideIds();
+
+            //sethdonohue - if activeSlideId attribute was set by implementation then we need to sync the activeSlideIndex for incrementing to work
+            if (this.activeSlideId) {
+                this.activeSlideIndex = this.slideIds.indexOf(this.activeSlideId);
+            } else {
+                this.activeSlideId = this.slideIds[this.activeSlideIndex] as string;
+            }
+
+            this.items.forEach((item: HTMLElement, index: number) => {
+                if (index === this.activeSlideIndex) {
+                    item.classList.add("active-slide");
+                    item.removeAttribute("hidden");
+                } else {
+                    item.setAttribute("hidden", "");
+                }
+                if (
+                    !item.getAttribute("aria-label") ||
+                    !item.getAttribute("aria-labelledby")
+                ) {
+                    item.setAttribute(
+                        "aria-label",
+                        `${index + 1} of ${this.items.length}`
+                    );
+                }
+
+                item.setAttribute(
+                    "id",
+                    item.getAttribute("id")
+                        ? (item.getAttribute("id") as string)
+                        : `slide-${index + 1}`
+                );
+
+                item.classList.add("slide");
+                // sethdonohue - per ARIA spec role=group and roledescription=slide must be on the slide container for basicPattern (not tabbed) implementation
+                item.setAttribute("role", "group");
+                item.setAttribute("aria-roledescription", "slide");
+
+                return item;
+            });
+        }
+    }
+
+    public handleFlipperClick(direction: 1 | -1, e: Event): void {
+        this.incrementSlide(direction);
+    }
+
+    public handleFlipperKeypress = (direction: 1 | -1, e: KeyboardEvent): void => {
+        switch (e.keyCode) {
+            case keyCodeSpace:
+                if (
+                    e.target !== this.nextFlipperDefault &&
+                    e.target !== this.previousFlipperDefault
+                ) {
+                    break;
+                }
+            case keyCodeEnter:
+                this.paused = true;
+                this.incrementSlide(direction);
+                break;
+        }
+    };
+
+    public handleDefaultFlipperKeypress = (direction: 1 | -1, e: KeyboardEvent): void => {
+        switch (e.keyCode) {
+            case keyCodeSpace:
+                this.paused = true;
+                this.incrementSlide(direction);
+                break;
+        }
+    };
+
+    public change = (): void => {
+        // sethdonohue - reference to carousel is passed for the author to get access to the paused, activeSlideId, and other states.
+        this.$emit("change", this.carousel);
```

Comment:
you don't need to provide `this.carousel` because that is a reference to the WC itself, which is just the `this` object in this context. The author can access the carousel by inspecting `event.target` to read any data about the carousel they wish to read.